examples/js/interactive_button.js

```diff
@@ -312,55 +311,88 @@ function loop() {
+            selectedField: option + "ControlSelected",
+            closestPoint: null,
+            closestPointDistance: Number.MAX_SAFE_INTEGER,
+        };
+    }
+
+    //Due to the nature of having 2 controllers, it's easier to determine if a
+    //component is idle by going through each ui element 1 by 1 using all
+    //raycasters. This way of adjusting state does not take the UI being behind
+    //another mesh into account, but does allow stacked component state changes
+    for(let i = 0; i < objsToTest.length; i++) {
+        const obj = objsToTest[i];
+        const component = obj.uiComponent; //Null for non-stateful components
+        const isStateful = statefulComponents.has(obj.uiComponent);
+        let raycasted = false;
+        for(let option in selectors) {
+            const selector = selectors[option];
+            const raycaster = selector['raycaster'];
+            const selected = control[selector['selectedField']];
+            let intersections;
+            if(raycaster == null) {
+                intersections = [];
+            } else {
+                intersections = raycaster.intersectObject(obj, true);
+            }
+            if(intersections.length != 0) {
+                const distance = intersections[0].distance;
+                if(distance < selector['closestPointDistance']) {
+                    selector['closestPointDistance'] = distance;
+                    selector['closestPoint'] = intersections[0].point;
+                }
+                raycasted = true;
+                if(isStateful) {
+                    if(selected) {
+                        if(!component.selectedOwners.has(option)) {
+                            component.selectedOwners.add(option);
+                        }
+                        if(component.currentState != "selected") {
+                            component.setState("selected");
+                        }
+                    } else {
+                        if(component.selectedOwners.has(option)) {
+                            component.selectedOwners.delete(option);
+                        }
+                        if(component.currentState != "hovered"
+                            && component.selectedOwners.size == 0)
+                        {
+                            component.setState("hovered");
+                        }
+                    }
+                }
+            } else if(isStateful && component.selectedOwners.has(option)) {
+                if(!selected) {
+                    component.selectedOwners.delete(option);
+                }
+            }
+        }
+        if(isStateful && !raycasted && component.selectedOwners.size == 0) {
+            component.setState("idle");
+        }
+    }
+
+    //Now we need to set the point for right and left controllers
+    updateControllerPoint('left', selectors['left']);
+    updateControllerPoint('right', selectors['right']);
 
```

Comment:
I don't know... It's a good thing if it fix the bug you mentioned, but the goal of the examples is to provide a clear overview so that a newcomer can understand how the core mechanics work. Now the core mechanics (component.setupState, component.setState...) got somewhat drowned in the layers of iterations, and it's less clear what's happening.

Looking at it, I think having a handful of control modules might finally be a better move. Especially if you consider that more examples with buttons and other interactions should come, in which all this raycaster part will have to be copy-pasted, in detriment of clarity and maintainability.

I still don't think the lib should have a modified THREE.Raycaster, it should always be usable with the last Three.js, but this raycaster could be used in separate modules, a bit like VRControl. However, VRControl could be split to only handle VR controllers raycasting, and the examples, when not detecting any VR controllers, would instead call MouseController module methods, instead of VRControl handling the fallback.